AWS-CodePipeline/two-environment-pipeline-template/{{cookiecutter.outputDir}}/codepipeline.yaml

```diff
@@ -0,0 +1,814 @@
+        # - Name: UnitTest
+        #   Actions:
+        #     - Name: UnitTest
+        #       ActionTypeId:
+        #         Category: Build
+        #         Owner: AWS
+        #         Provider: CodeBuild
+        #         Version: "1"
+        #       Configuration:
+        #         ProjectName: !Ref CodeBuildProjectUnitTest
+        #       InputArtifacts:
+        #         - Name: SourceCodeAsZip
+
+        - !If
+          - IsFeatureBranchPipeline
+          - Name: BuildAndDeployFeatureStack
+            Actions:
+              - Name: CodeBuild
+                ActionTypeId:
+                  Category: Build
+                  Owner: AWS
+                  Provider: CodeBuild
+                  Version: "1"
+                Configuration:
+                  ProjectName: !Ref CodeBuildProjectBuildAndDeployFeature
+                InputArtifacts:
+                  - Name: SourceCodeAsZip
+          - !Ref AWS::NoValue
+
+        - !If
+          - IsMainBranchPipeline
+          - Name: BuildAndPackage
+            Actions:
+              - Name: CodeBuild
+                ActionTypeId:
+                  Category: Build
+                  Owner: AWS
+                  Provider: CodeBuild
+                  Version: "1"
+                Configuration:
+                  ProjectName: !Ref CodeBuildProjectBuildAndPackage
+                InputArtifacts:
+                  - Name: SourceCodeAsZip
+                OutputArtifacts:
+                  - Name: BuildArtifactAsZip
+          - !Ref AWS::NoValue
+        
+        # Uncomment the following step for running the integration tests
+        # - !If
+        #   - IsMainBranchPipeline
+        #   - Name: DeployTest
+        #     Actions:
+        #       - Name: DeployTest
+        #         ActionTypeId:
+        #           Category: Build
+        #           Owner: AWS
+        #           Provider: CodeBuild
+        #           Version: "1"
+        #         Configuration:
+        #           ProjectName: !Ref CodeBuildProjectDeploy
+        #           EnvironmentVariables: !Sub |
+        #             [
+        #               {"name": "ENV_TEMPLATE", "value": "packaged-test.yaml"},
+        #               {"name": "ENV_REGION", "value": "${TestingRegion}"},
+        #               {"name": "ENV_STACK_NAME", "value": "${TestingStackName}"},
+        #               {"name": "ENV_PIPELINE_EXECUTION_ROLE", "value": "${TestingPipelineExecutionRole}"},
+        #               {"name": "ENV_CLOUDFORMATION_EXECUTION_ROLE", "value": "${TestingCloudFormationExecutionRole}"},
+        #               {"name": "ENV_BUCKET", "value": "${TestingArtifactBucket}"},
+        #               {"name": "ENV_IMAGE_REPOSITORY", "value": "${TestingImageRepository}"}
+        #             ]
+        #         InputArtifacts:
+        #           - Name: BuildArtifactAsZip
+        #         RunOrder: 1
```

Comment:
Major: you merged the DeployTest stage inside the integration-test by mistake.

Reply:
good catch!